app/uk/gov/hmrc/agentsubscriptionfrontend/validators/CommonValidators.scala

```diff
@@ -204,19 +204,16 @@ object CommonValidators {
       }
   }
 
-  private def isUtrValid(utrStr: String): Boolean = TaxIdentifierFormatters.normalizeUtr(utrStr).nonEmpty
-
   private def utrConstraint(errorMessages: UtrErrors = DefaultUtrErrors): Constraint[String] = Constraint[String] {
     fieldValue: String =>
       val formattedField = fieldValue.replace(" ", "")
       val (blank, invalid) = errorMessages
 
       Constraints.nonEmpty(formattedField) match {
         case _: Invalid => Invalid(ValidationError(blank))
-        case _ if formattedField.map(_.isDigit).reduce(_ && _) && formattedField.size != UtrMaxLength =>
+        case _ if !formattedField.map(_.isDigit).reduce(_ && _) || formattedField.size != UtrMaxLength =>
```

Comment:
Could use `formattedField.forall(_.isDigit)`
This check is used in TaxIdentifierFormatters.scala too, if it's something that can given a snappy name, it could be made into a function that helps make the code read easily in these two places.